packages/hub/searchers.js

```diff
@@ -73,8 +73,10 @@ class Searchers {
         upstreamDoc: { data: resource, meta },
         read: this._read(branch)
       });
-      let pristineResult = await (documentContext.pristineDoc());
-      authorizedResult = await schema.applyReadAuthorization(pristineResult, { session, type, id });
+      let pristineResult = await documentContext.pristineDoc();
+      if (pristineResult) {
+        authorizedResult = await schema.applyReadAuthorization(pristineResult, { session, type, id });
```

Comment:
This makes me see a refactor we should do.

`applyReadAuthorization` would be simpler if we moved it out of Schema and into DocumentContext.

The DocumentContext already has access to schema, session, type, and id, and it's already configured with the ability to `read` other documents, which would mean we could eliminate the new injections you needed in Schema.

Reply:
We could do that but one of the auxiliary methods, `_readAuthorizationForPermissions` uses `this.searchers` and `this.controllingBranch` that are accessed via DI and I don't think are available in DocumentContext. So they'd have to be injected there, too.

https://github.com/cardstack/cardstack/pull/421/files#diff-ed3aae8cfdd1353fe73e9430bea36cd8R476

Not saying I cannot do that, I'm just still not very knowledgeable about what should go where so I'm bringing this up because maybe the refactoring loses some of its rationale (its raison d'être, if I want to be a snob).

Reply:
The other tasks you pointed out are completed.

Reply:
Oh, wait, I glossed over one of your sentences ("and it's already configured with the ability to read other documents"), I'll take a look at how that works.